evalml/tests/automl_tests/test_automl.py

```diff
@@ -5047,6 +5054,62 @@ def test_default_algorithm_uses_n_jobs(X_y_binary, AutoMLTestEnv):
     assert n_checked and n_feature_selector_checked
 
 
+@pytest.mark.parametrize("input_type", ["pd", "ww"])
+@pytest.mark.parametrize("automl_algorithm", ["default", "iterative"])
+@pytest.mark.parametrize("problem_type", ProblemTypes.all_problem_types)
+def test_exclude_featurizers(
+    automl_algorithm,
+    problem_type,
+    input_type,
+    get_test_data_from_configuration,
+    AutoMLTestEnv,
+):
+    parameters = {}
+    if is_time_series(problem_type):
+        parameters = {
+            "time_index": "dates",
+            "gap": 1,
+            "max_delay": 1,
+            "forecast_horizon": 1,
+        }
+
+    X, y = get_test_data_from_configuration(
+        input_type, problem_type, column_names=["dates", "text", "email", "url"]
+    )
+
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type=problem_type,
+        problem_configuration=parameters,
+        automl_algorithm=automl_algorithm,
+        exclude_featurizers=[
+            "DatetimeFeaturizer",
+            "EmailFeaturizer",
+            "URLFeaturizer",
+            "NaturalLanguageFeaturizer",
+            "TimeSeriesFeaturizer",
+        ],
+    )
+
+    env = AutoMLTestEnv(problem_type)
+    with env.test_context(score_return_value={automl.objective.name: 1.0}):
+        automl.search()
+
+    pipelines = [
+        automl.get_pipeline(i) for i in range(len(automl.results["pipeline_results"]))
+    ]
+
+    # A check to make sure we actually retrieve constructed pipelines from the algo.
+    assert len(pipelines) > 0
+
+    assert not any([DateTimeFeaturizer.name in pl for pl in pipelines])
```

Comment:
`get_pipeline()` returns a pipeline instance so you would need to check for `DateTimeFeaturizer.name in pl.component_graph.compute_order for pl in pipelines`. You could also do `DateTimeFeaturizer.name in pl.name for pl in pipelines` but the compute order check is more explicit as `compute_order` is the list of components in text.

Reply:
Thanks for the clarification.

Updated: [f111676](https://github.com/alteryx/evalml/pull/3631/commits/f111676fb3aae55b9fb37fde1a05660a22ba8e3d)